nix/shells.nix

```diff
@@ -7,94 +7,47 @@
 , holonix
 , hcRustPlatform
 , hcToplevelDir
-, hcTargetPrefixEval
+, nixEnvPrefixEval
 , pkgs
 }:
 
 let
-  inherit (lib.attrsets) mapAttrsToList mapAttrs;
+  hcMkShell = input: mkShell {
+    # mkShell reverses the inputs list, which breaks order-sensitive shellHooks
+    inputsFrom = lib.reverseList [
+      { shellHook = nixEnvPrefixEval; }
 
-  commonShellHook = ''
-    ${hcTargetPrefixEval}
-    echo Using "$HC_TARGET_PREFIX" as target prefix...
+      holonix.shell
 
-    export CARGO_TARGET_DIR="$HC_TARGET_PREFIX/target"
-    export CARGO_CACHE_RUSTC_INFO=1
-    export CARGO_HOME="$HC_TARGET_PREFIX/.cargo"
-    export CARGO_INSTALL_ROOT="$HC_TARGET_PREFIX/.cargo"
-    # FIXME: we currently rely on lair-keystore being installed and found here by `holochain`
-    export PATH="$PATH:$CARGO_INSTALL_ROOT/bin"
+      { shellHook = ''
+        echo Using "$NIX_ENV_PREFIX" as target prefix...
 
-    export HC_TEST_WASM_DIR="$HC_TARGET_PREFIX/.wasm_target"
-    mkdir -p $HC_TEST_WASM_DIR
+        export HC_TEST_WASM_DIR="$NIX_ENV_PREFIX/.wasm_target"
+        mkdir -p $HC_TEST_WASM_DIR
 
-    export HC_WASM_CACHE_PATH="$HC_TARGET_PREFIX/.wasm_cache"
-    mkdir -p $HC_WASM_CACHE_PATH
+        export HC_WASM_CACHE_PATH="$NIX_ENV_PREFIX/.wasm_cache"
```

Comment:
@thedavidmeister I wonder if this should rather be inside `${CARGO_TARGET_DIR}`, WDYT?